src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -56,167 +53,40 @@ function createAndAccumulateChangeEvent(inst, nativeEvent, target) {
   // deleting text, so we ignore its input events.
-  isInputEventSupported =
-    isEventSupported('input') &&
-    (!document.documentMode || document.documentMode > 9);
-}
-
-/**
- * (For IE <=9) Starts tracking propertychange events on the passed-in element
- * and override the value property so that we can distinguish user events from
- * value changes in JS.
- */
-function startWatchingForValueChange(target, targetInst) {
-  activeElement = target;
-  activeElementInst = targetInst;
-  activeElement.attachEvent('onpropertychange', handlePropertyChange);
-}
-
-/**
- * (For IE <=9) Removes the event listeners from the currently-tracked element,
- * if any exists.
- */
-function stopWatchingForValueChange() {
-  if (!activeElement) {
-    return;
-  }
-  activeElement.detachEvent('onpropertychange', handlePropertyChange);
-  activeElement = null;
-  activeElementInst = null;
-}
-
-/**
- * (For IE <=9) Handles a propertychange event, sending a `change` event if
- * the value of the active element has changed.
- */
-function handlePropertyChange(nativeEvent) {
-  if (nativeEvent.propertyName !== 'value') {
-    return;
-  }
-  if (getInstIfValueChanged(activeElementInst)) {
-    manualDispatchChangeEvent(nativeEvent);
-  }
+  isTextInputEventSupported =
+    !document.documentMode || document.documentMode > 9;
 }
 
-function handleEventsForInputEventPolyfill(topLevelType, target, targetInst) {
-  if (topLevelType === 'topFocus') {
-    // In IE9, propertychange fires for most input events but is buggy and
-    // doesn't fire when text is deleted, but conveniently, selectionchange
-    // appears to fire in all of the remaining cases so we catch those and
-    // forward the event if the value has changed
-    // In either case, we don't want to call the event handler if the value
-    // is changed from JS so we redefine a setter for `.value` that updates
-    // our activeElementValue variable, allowing us to ignore those changes
-    //
-    // stopWatching() should be a noop here but we call it just in case we
-    // missed a blur event somehow.
-    stopWatchingForValueChange();
-    startWatchingForValueChange(target, targetInst);
-  } else if (topLevelType === 'topBlur') {
-    stopWatchingForValueChange();
-  }
-}
-
-// For IE8 and IE9.
+// In IE9 the input event does not fire for deleting text. Luckily
+// selectionchange does, so we also listen for that in those cases.
 function getTargetInstForInputEventPolyfill(topLevelType, targetInst) {
+  // On the selectionchange event, the target is the document which isn't
+  // helpful becasue we need the input, so we use the activeElement instead.
+  const eventInst = topLevelType === 'topSelectionChange'
+    ? getActiveElement()
+    : targetInst;
+
   if (
-    topLevelType === 'topSelectionChange' ||
-    topLevelType === 'topKeyUp' ||
-    topLevelType === 'topKeyDown'
+    topLevelType === 'topInput' ||
+    topLevelType === 'topChange' ||
+    topLevelType === 'topSelectionChange'
```

Comment:
the original comment:

>  99% of the time, keydown and keyup aren't necessary. IE8 fails to fire 
> propertychange on the first input event after setting `value` from a		
> script and fires only keydown, keypress, keyup. Catching keyup usually		
> gets it and catching keydown lets us fire an event for the first		
> keystroke if user does a key repeat (it'll be a little delayed: right		
> before the second keystroke). Other input methods (e.g., paste) seem to		
> fire selectionchange normally.

It says IE8 only but one never knows right? We could add them back for sense of security, downside is just a few more lines of code, and maybe cargo-culting in code.